lib/server.js

```diff
@@ -9,6 +9,39 @@ var Hapi = require('hapi')
 
 var HEX_STRING = require('./routes/validators').HEX_STRING
 
+function trimLocale(header) {
+  if (!header) {
+    return header
+  }
+  if (header.length < 256) {
+    return header.trim()
+  }
+  var parts = header.split(',')
+  var str = parts[0]
+  if (str.length >= 255) { return null }
+  for (var i = 1; i < parts.length && str.length + parts[i].length < 255; i++) {
+    str += ',' + parts[i]
+  }
+  return str.trim()
+}
+
+function logEndpointErrors(response, log) {
+  // When requests to DB timeout and fail for unknown reason they are an 'EndpointError'.
+  // The error response hides error information from the user, but we log it here
+  // to better understand the DB timeouts.
```

Comment:
hmm maybe, cases against the suggested though:

`DatabaseError` -> not exactly because customs server requests also use a pool, there is no "Database" there

`TimeoutError` -> not exactly because the pool can throw for different reasons.

It could be `RequestError` ?